### ISSUE TITLE
Lots of UnusedImports false positives in 1.21.0-RC1 with type resolution

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#4955]
Updating from 1.20.0 to 1.21.0-RC1, if I run `gradle detektRelease` I get a bunch of `UnusedImports` false positives.

There's seemingly no commonality between the false positives; some of them are inline functions, some are generic functions, some are interfaces, etc...

**Comment by cortinico** [comment]
Can you provide a reproducer?

**Comment by eygraber** [comment]
I can't share with this specific project, so I'll try some of my public ones.

**Comment by eygraber** [comment]
I haven't been able to repro in other projects. I'll see if I can get a shareable subset of my main project to repro.

**Comment by eygraber** [comment]
It looks like the issue was gradle build/configuration cache related. Building without them fixed the issue.

**Comment by scottkennedy** [comment]
I'm seeing this too.  Is this not considered a Detekt bug?  Should it be disabled by default?

Otherwise, how can I disable it?

**Comment by BraisGabin** [comment]
Which kotlin version are you using? This seems like an issue on kotlin/gradle that something related with detekt.

**Comment by scottkennedy** [comment]
Kotlin 1.6.0, Gradle 7.4.2.

**Comment by scottkennedy** [comment]
Okay I have one example of this in 1.21.0.

Kotlin 1.7.10, Gradle 7.5.

I unfortunately can't post the whole file, nor am I able to get a smaller reproduction.

But it's something like:
```kt
fun getSomething(id: UUID): List<UnusedImport>? {
    return something
}
```

That `UnusedImport` isn't used anywhere else in the file.
